pending-documentation.md

```diff
@@ -107,6 +107,143 @@ argument and parameter are K, and it has no effect on the resulting sort of
 the term. However, it will nevertheless affect the kore generated from the term
 by introducing an additional parameter to the symbol generated for the term.
 
+### `strict` and `seqstrict` attributes
+
+The strictness attributes allow defining evaluation strategies without having to
+explicitely make rules which implement them. This is done by injecting *heating*
+and *cooling* rules for the subterms. For this to work, you need to define what
+a *result* is for K, by extending the  `KResult` sort.
+
+For example:
+
+```k
+syntax AExp ::= Int
+              | AExp "+" AExp [strict]
+```
+
+This generates two heating rules (where the hole syntax `[]` is automatically
+added to create an evaluation context):
+
+```k
+rule <k> AE1:AExp + AE2:AExp => AE1 ~> [] + AE2 ... </k> requires notBool isKResult(AE1)
+rule <k> AE1:AExp + AE2:AExp => AE2 ~> AE1 + [] ... </k> requires notBool isKResult(AE2)
+```
+
+And two corresponding cooling rules:
+
+```k
+rule <k> AE1:KResult ~> [] + AE2 => AE1 + AE2 ... </k>
```

Comment:
@ttuegel needs to speak about the haskell backend and the isKResult predicate, but I know the llvm backend does not generate isKResult for cooling rules.

Reply:
The `definition.kore` generated for the Haskell backend contains function definitions for `isKResult`.